### PR TITLE
[file_system_http_cache] Add more features to API, and remove some not-implemented tags

### PR DESCRIPTION
**Pull request by ravenblackx**
Commit Message: [file_system_http_cache] Add more features to API, and remove some not-implemented tags
Additional Description:
  Add API for:
  * `evict_fraction`: evicting a fraction of the exceeded limits extra to reduce churn.
  * `max_eviction_frequency`: do an eviction pass before it's necessary, to ensure the stats aren't too out of date.
  * `min_eviction_frequency`: don't do an eviction pass even if limits are reached, to reduce churn.
  * `create_cache_path`: option to attempt to create the cache path if it does not exist.

  Remove not-implemented-hide for:
  * `max_cache_size_bytes`
  * `max_cache_entry_count`
  
  Clarify name for:
  * `max_cache_entry_size_bytes` -> `max_individual_cache_entry_size_bytes`: to avoid confusion with `max_cache_size_bytes`

Risk Level: None, extension is WIP and no change to existing behavior.
Testing: n/a
Docs Changes: Adds some generated docs.
Release Notes: n/a
Platform Specific Features: n/a

### CODE REVIEW

File: api/envoy/extensions/http/cache/file_system_http_cache/v3/file_system_http_cache.proto
Comment (htuch) [inline api/envoy/extensions/http/cache/file_system_http_cache/v3/file_system_http_cache.proto:103]: Technically it's a period, the reciprocal is frequency.

**Comment by htuch** [comment]
@jmarantz @toddmgreer can you folks provide a domain specific review?

File: api/envoy/extensions/http/cache/file_system_http_cache/v3/file_system_http_cache.proto
Comment (ravenblackx) [reply api/envoy/extensions/http/cache/file_system_http_cache/v3/file_system_http_cache.proto]: Good call. Updated. (Though "every 10 minutes" is a frequency, there's no way to phrase that as a duration and variable name.)

**Comment by RyanTheOptimist** [comment]
@jmarantz friendly maintainer on-call ping.

**Review by jmarantz — APPROVED** [review]
This looks great, thanks!

File: api/envoy/extensions/http/cache/file_system_http_cache/v3/file_system_http_cache.proto
Comment (htuch) [inline api/envoy/extensions/http/cache/file_system_http_cache/v3/file_system_http_cache.proto:100]: I'm a little unclear after reading this what help there is from an eviction pass not triggered by cache limits. There's something above about syncing on-disk state, but I think this could be clearer.

Comment (ravenblackx) [reply api/envoy/extensions/http/cache/file_system_http_cache/v3/file_system_http_cache.proto:104]: Added a top-level link to the design doc (which isn't actually there yet, but the [PR that adds it](https://github.com/envoyproxy/envoy/pull/24642) and this one are racing for who can be the slowest!)
Also elaborated with an example.

**Comment by jmarantz** [comment]
I think I am good with this and this one is just waiting for @htuch API review and he can just merge.

**Review by htuch — APPROVED** [review]
Thanks1

**Pull request merged**
